builder/virtualbox/common/step_attach_guest_additions.go

```diff
@@ -77,7 +77,7 @@ func (s *StepAttachGuestAdditions) Cleanup(state multistep.StateBag) {
 		"--medium", "none",
 	}
 
-	if err := driver.VBoxManage(command...); err != nil {
-		ui.Error(fmt.Sprintf("Error unregistering guest additions: %s", err))
-	}
+	// Remove the ISO. Note that this will probably fail since
```

Comment:
> oh, you mean the debug logs from the driver.vboxmanage?

Right! Apologies for not making that obvious above. An example of what I'm on about can be seen in the log below. Clearly the first VBoxManage command is from step_remove_devices while the next one (with the error message) is from the clean up in step_attach_guest_additions.

```
2017/01/30 17:10:21 ui: ==> virtualbox-iso: Gracefully halting virtual machine...
...
2017/01/30 17:10:23 packer: 2017/01/30 17:10:23 VM shut down.
2017/01/30 17:10:23 packer: 2017/01/30 17:10:23 Executing VBoxManage: []string{"storageattach", "centos-7-x86_64", "--storagectl", "IDE Controller", "--port", "0", "--device", "1", "--medium", "none"}
2017/01/30 17:10:23 packer: 2017/01/30 17:10:23 stdout:
2017/01/30 17:10:23 packer: 2017/01/30 17:10:23 stderr:
...
2017/01/30 17:10:24 ui: ==> virtualbox-iso: Preparing to export machine...
...
...
2017/01/30 17:11:02 packer: 2017/01/30 17:11:02 Executing VBoxManage: []string{"storageattach", "centos-7-x86_64", "--storagectl", "IDE Controller", "--port", "0", "--device", "1", "--medium", "none"}
2017/01/30 17:11:02 packer: 2017/01/30 17:11:02 stdout:
2017/01/30 17:11:02 packer: 2017/01/30 17:11:02 stderr: VBoxManage: error: No storage device attached to device slot 1 on port 0 of controller 'IDE Controller'
2017/01/30 17:11:02 packer: VBoxManage: error: Details: code VBOX_E_OBJECT_NOT_FOUND (0x80bb0001), component SessionMachine, interface IMachine, callee nsISupports
2017/01/30 17:11:02 packer: VBoxManage: error: Context: "DetachDevice(Bstr(pszCtl).raw(), port, device)" at line 393 of file VBoxManageStorageController.cpp
2017/01/30 17:11:02 ui: ==> virtualbox-iso: Unregistering and deleting virtual machine...
...
```

It's only *after* having poked around in the source code that I now know where this error is coming from, and that it can safely be ignored. However, when I was looking at this before (from the standpoint of an end user) the message worried me - What was going wrong to cause this message? I guess this is about end user perception...

My concern is that at the end of a Packer build there is the possibility of a bunch of errors appearing in the logs and no clear indication of where they are coming from or that they are safe to ignore. This is why I was hoping we could somehow avoid running the detach command again in the clean up phase.